### PR TITLE
ProjectMetadataResponse permission

### PR DESCRIPTION
**Pull request by tom114**
## Description of changes
Created a class called `ProjectMetadataResponse`.  This class groups together metadata from samples within a project, and the project itself.  This is required so that spring security can do its checks on the return values to ensure the requesting user has access to the metadata within the context of that project.

This also added a permission to the methods requesting a `ProjectMetadataResponse`.  This will ensure that the user requesting metadata has access to all the `MetadataTemplateField`s in the returned response.

This PR removed `MetadataFieldResponse.java` because it added extra complexity that wasn't really needed.

## Related issue
Related to #931

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
* [x] Tests added (or description of how to test) for any new features.
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/security/permissions/metadata/ReadProjectMetadataResponsePermission.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		 */
+		boolean allFieldsValid = fields.stream()
+				.filter(field -> {
+					//if we have a restriction on a field, compare it against the user's role on the project
+					if (restrictionMap.containsKey(field)) {
+						MetadataRestriction metadataRestriction = restrictionMap.get(field);
+						ProjectRole restrictionRole = metadataRestriction.getLevel();
+
+						//compare the restriction level to the user's role
+						if (userProjectRole.getLevel() >= restrictionRole.getLevel()) {
```
Comment (joshsadam) [inline src/main/java/ca/corefacility/bioinformatics/irida/security/permissions/metadata/ReadProjectMetadataResponsePermission.java:104]: Couldn't this be simplified to just 

```java
userProjectRole.getLevel() <= restrictionRole.getLevel()
```

Comment (tom114) [reply src/main/java/ca/corefacility/bioinformatics/irida/security/permissions/metadata/ReadProjectMetadataResponsePermission.java]: Well sure if you don't like giant unnecessary `if` blocks.  Simplified in 8ee46ca3a5

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me

**Review by joshsadam — APPROVED** [review]
I suppose this looks great, thanks~

**Pull request merged**